Fix: fix side user profile card

i-nooo-2 opened the pull request:
<!-- PR 제목은 커밋 메세지 컨벤션 형식으로 작성 -->

## 🏷️ 이슈 번호 

- close #60 

## 🧱 작업 사항
Side User Profile Card에 사용자의 이미지를 받아 메타 데이터를 수정 및 삭제하는 기능을 추가하였습니다.

## 📸 결과물
![image](https://github.com/user-attachments/assets/a24f2f6a-b296-4707-af5f-70124512282e)
![image](https://github.com/user-attachments/assets/cc7d9d63-5703-4a3b-a42e-2a6c60cd858b)
![image](https://github.com/user-attachments/assets/210dd251-8082-4d3c-a85d-ac7e34b67573)

## 💬 공유 포인트 및 논의 사항
구현 시 모든 기능이 작동되는 것을 확인하였으나, 구현 완료 후 서버에러(500)가 발생하여 추가적인 에러 테스트를 진행하지 못 하였습니다.. 때문에 추가적인 버그 잡기가 필요합니다. 또한 코드 리팩토링이 필요합니다.

i-nooo-2 committed:
feat : profile api 연동

다만 기존부터 아이콘을 받아 출력할 수 있게 되어있었기 때문에, 이번 구현이 적절한 것인지 모르겠음

i-nooo-2 committed:
fix: resevation icon svg 교체

i-nooo-2 committed:
Feat: 프로필 이미지 업데이트 및 삭제 기능

eunbinnie reviewed (commented):
1. app/page.tsx는 삭제되는 게 맞습니다!

2. _types/user.ts 에서 GetUserType, UserProfileData가 타입명만 다르고 내용은 아예 다 똑같아서요 하나로 통일하고, 다른 하나는 삭제 부탁드립니다! 그리고 삭제한 타입 쓰고있는 파일들 검색해서 바꿔주시고 머지해주세요 빌드 꼭 부탁드립니다!

3. 예약 내역 페이지로 갔는데 사이드 프로필은 여전히 내 정보가 활성화되어있습니다
![image](https://github.com/user-attachments/assets/787e42a4-f8d8-4a18-b80a-07531c0ceac3)

4. 내 체험 관리 href 등록 부탁드립니다
{
    icon: <IconManageMyActivity />,
    title: '내 체험 관리',
    href: '#',
  },

5. 모바일에서는 이미지가 안뜹니다! 확인 부탁드립니다! 그리고 모바일에서는 마이페이지 url이 /my라서 아무것도 활성화안되는 게 맞습니다. 현재는 내 정보가 활성화되어있습니다
![image](https://github.com/user-attachments/assets/d27d82d8-24a4-4eb1-ad00-e40376feebe2)

6. 그리고 사이드 스티키 레이아웃에서 사이드 유저 프로필을 조건부 렌더링하고 있어서 스켈레톤이 필요합니다. 스켈레톤을 쓰지 않으려면 으음 css로 none block 처리해야할 것 같습니다.

7. ProfileButton a태그 말고 Link 태그로 바꿔주세요

8. 이미지 등록이 사이트에서 안되는 것 같은데 스웨거에서는 됩니다 확인부탁드립니다

i-nooo-2 committed:
Merge branch 'develop' into fix/#60-fix-side-user-profile-card

hongseoha commented on app/_types/user.ts:
```diff
@@ -0,0 +1,8 @@
+export interface UserProfileData {
```
해당부분 develop과 합쳤을 때 conflict 발생합니다! GetUserType으로 이름 바꿔주시면 될 것 같아요.

hongseoha commented on app/_components/side-user-profile-card/_components/ChangeProfileImageModal/index.tsx:
```diff
@@ -0,0 +1,102 @@
+          </button>
+        </div>
+
+        {!showDeleteConfirmation ? (
+          <>
+            <h3 className="pb-4 text-center">프로필 이미지 변경</h3>
+            <div className="mb-5 size-[160px] overflow-hidden rounded-[50%]">
+              <Image src={currentProfileImage} alt="current avatar" width={160} height={160} priority />
+            </div>
+            <div className="flex w-full content-around items-center justify-between">
+              <button type="button" onClick={handleDeleteClick} className="text-base">
+                이미지 삭제
+              </button>
+              <label className="block cursor-pointer">
+                <input type="file" accept="image/*" onChange={handleImageUpload} className="hidden" />
+                <span className="rounded bg-green-200 px-4 py-2 text-base text-white">이미지 업로드</span>
+              </label>
+            </div>
```
![image](https://github.com/user-attachments/assets/89fe68c2-f108-45cc-9f73-2d2d800757a9)

모바일에서 버튼 사이즈 조정이 필요할 것 같습니다.

hongseoha reviewed (commented):
수고하셨습니다. 은빈님 코멘트 외에 다른 부분 조금 추가했습니다. 확인 부탁드려요~

i-nooo-2 committed:
Fix: page 변동에 따라 profile card btn selected도 변동

- page 변동에 따라 profile card btn selected도 변동
- app/page.tsx 삭제 (마이페이지에서 테스트 및 추가 개발 진행)

i-nooo-2 committed:
Chore: Shadcn ui component 환경 구성

i-nooo-2 committed:
Feat: 스켈레톤 적용

i-nooo-2 committed:
Fix: 모바일 대응, default profile image를 svg로 변경

i-nooo-2 committed:
Fix: Api 500 에러 해결

i-nooo-2 committed:
Feat: Full page spinner 추가

i-nooo-2 committed:
Fix: 모바일 대응

i-nooo-2 committed:
Fix: 레이어 순서 문제 해결

i-nooo-2 committed:
Merge branch 'develop' into fix/#60-fix-side-user-profile-card

i-nooo-2 committed:
Fix: z-index 수정

i-nooo-2 committed:
Fix: Profile Img 없을 때 무한 루프 도는 이슈

i-nooo-2 committed:
Fix: Profile Img 없을 경우 검사 후 대체 이미지 적용

i-nooo-2 committed:
Fix: Image 태그의 svg 출력 문제

Image 태그는 svg 이미지 파일을 받아오는 url의 경우에는 호스팅이 되지 않았습니다. 때문에 초기화 이미지를 올릴 때 svg를 png로 변환하는 과정을 거쳤습니다.

convertSVGToPNGFile();

edhcoding reviewed (approved):
기능 잘 동작합니다! 고생하셨습니다 ㅎㅎ
